rotkehlchen/db/dbhandler.py

```diff
@@ -2389,11 +2389,17 @@ def get_trades(self, cursor: 'DBCursor', filter_query: TradesFilterQuery, has_pr
 
         return trades
 
-    def delete_trade(self, write_cursor: 'DBCursor', trade_id: str) -> Tuple[bool, str]:
-        write_cursor.execute('DELETE FROM trades WHERE id=?', (trade_id,))
-        if write_cursor.rowcount == 0:
-            return False, 'Tried to delete non-existing trade'
-        return True, ''
+    def delete_trades(self, write_cursor: 'DBCursor', trades_ids: List[str]) -> None:
+        """Removes trades from the database using their `trade_id`.
+        May raise:
+        - InputError if any of the `trade_id` are non-existent.
+        """
+        write_cursor.executemany(
+            'DELETE FROM trades WHERE id=?',
+            [(trade_id,) for trade_id in trades_ids],
+        )
+        if write_cursor.rowcount != len(trades_ids):
+            raise InputError('Tried to delete one or more non-existing trade(s)')
```

Comment:
At this moment you have already deleted the existing entries but you are returning an error. I would expect it to fail but rollback or return some kind of different error. What do you think? Maybe discuss it in discord? I'm not really sure about this one

Reply:
Okey, cleared on discord

Reply:
Discussed in Discord already.

Reply:
Yes, is that. The context gets the exception and then rollbacks the change. I was expecting to see it in the dbhandler file but happens in the rest.py when you exit the context there.